### ISSUE TITLE
Non-existant extensions raise wrong error

### ISSUE THREAD

**Issue by waylan** [Python-Markdown/markdown#105]
When an extension name is passed in as text, and no extension by that name exists, the wrong error is raised. When passing in `"bad_name"`, we're getting:

```
ValueError: Extension "__builtin__.NoneType" must be of type: "markdown.Extension".
```

when we should be getting a warning:

```
Failed loading extension bad_name from 'markdown.extensions.bad_name' or 'mdx_bad_name'. 
```

This is probably related to the changes made in commit 8bc4135947d9a2fc1749ebc4a8cdbd4d72d7a74e. Thanks to Italo Maia for the report.

**Comment by mitya57** [comment]
The commit message says "Non-existant extensions fail with a warning only", but I still get the traceback and ImportError:

```
>>> import markdown
>>> markdown.Markdown(['bad_name'])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "markdown/__init__.py", line 139, in __init__
    configs=kwargs.get('extension_configs', {}))
  File "markdown/__init__.py", line 165, in registerExtensions
    ext = self.build_extension(ext, configs.get(ext, []))
  File "markdown/__init__.py", line 203, in build_extension
    module = __import__(module_name_old_style)
ImportError: Failed loading extension 'bad_name' from 'markdown.extensions.bad_name' or 'mdx_bad_name'
```

I believe that the object still should still be created, and the bad extension should be skipped...

**Comment by waylan** [comment]
After making the above commit, we then received the report found in #112. As it turns out, our test suite has been expecting an error (rather than a warning) for this for years. We shouldn't be changing this now, so the change was reverted (in 5b3e724fb78da73ab87fb34e4ac9d9299773cfed) and the error returned was cleaned up to better reflect the actual problem (an ImportError rather than a confusing ValueError).

For the record, I made the first change based on comments in the code. Of course, the code didn't actually match the comments so that was wrong. I apparently failed to run the test suite after making this change or I never would have committed it. The current behavior is correct.
